File: analysis/copy_to_app.js
const fs = require('fs');
const path = require('path');

function copyOutputs() {
  const baseDir = path.dirname(__dirname);
  const srcDir = path.join(baseDir, 'analysis', 'outputs');
  const destDir = path.join(baseDir, 'app', 'lib', 'data');

  if (!fs.existsSync(destDir)) {
    fs.mkdirSync(destDir, { recursive: true });
  }

  const files = ['driver_features.json', 'vehicle_features.json', 'trip_features.json', 'fleet_summary.json', 'methodology.json'];

  console.log(`Copying files from ${srcDir} to ${destDir}...`);
  
  files.forEach(file => {
    const srcPath = path.join(srcDir, file);
    const destPath = path.join(destDir, file);

    if (fs.existsSync(srcPath)) {
      fs.copyFileSync(srcPath, destPath);
      console.log(`  Copied ${file}`);
    } else {
      console.warn(`  [Warning] Source file ${file} not found at ${srcPath}`);
    }
  });

  console.log('=== DATA DEPLOYMENT TO APP COMPLETE ===');
}

copyOutputs();

File: analysis/copy_to_app.py
import os
import shutil

def copy_outputs_to_app():
    base_dir = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
    src_dir = os.path.join(base_dir, 'analysis', 'outputs')
    dest_dir = os.path.join(base_dir, 'app', 'lib', 'data')
    
    os.makedirs(dest_dir, exist_ok=True)
    
    files = ['driver_features.json', 'vehicle_features.json', 'trip_features.json', 'fleet_summary.json', 'methodology.json']
    
    print(f"Copying files from {src_dir} to {dest_dir}...")
    for filename in files:
        src_path = os.path.join(src_dir, filename)
        dest_path = os.path.join(dest_dir, filename)
        
        if os.path.exists(src_path):
            shutil.copy(src_path, dest_path)
            print(f"  Copied {filename}")
        else:
            print(f"  [Warning] Source file {filename} not found.")
            
    print("=== DATA DEPLOYMENT TO APP COMPLETE ===")

if __name__ == '__main__':
    copy_outputs_to_app()
